### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by modmuss50**

### CODE REVIEW

File: enigma-swing/build.gradle [hunk 1]
```diff
@@ -1,23 +1,36 @@
...
 	id 'application'
-	id 'com.github.johnrengelman.shadow' version '7.0.0'
+	id 'com.github.johnrengelman.shadow'
 }
 
+def flatLafNatives = [
+        "windows-arm64@dll",
+		"windows-x86@dll",
+		"windows-x86_64@dll",
+		"linux-x86_64@so",
```
Comment (Juuxel) [inline enigma-swing/build.gradle:10]:
```suggestion
		"windows-arm64@dll",
		"windows-x86@dll",
		"windows-x86_64@dll",
		"linux-x86_64@so",
```

File: enigma-swing/build.gradle [hunk 2]
```diff
@@ -1,23 +1,36 @@
...
 	implementation project(':enigma')
 	implementation project(':enigma-server')
 
 	implementation 'net.sf.jopt-simple:jopt-simple:6.0-alpha-3'
-	implementation 'com.formdev:flatlaf:2.6'
-	implementation 'com.formdev:flatlaf-extras:2.6' // for SVG icons
-	implementation 'de.sciss:syntaxpane:1.2.1'
+	implementation 'com.formdev:flatlaf:3.2.1'
+	implementation 'com.formdev:flatlaf-extras:3.2.1' // for SVG icons
+	implementation 'de.sciss:syntaxpane:1.3.0'
```
Comment (ix0rai) [inline enigma-swing/build.gradle:20]:
```suggestion
	implementation 'de.sciss:syntaxpane:1.2.1'
```
syntaxpane 1.3 breaks scrolling line numbers, as I discovered when trying to update for quilt enigma. you'll have to either:
- stay on 1.2 and wait for a fix
- move to [syntaxpain](https://github.com/quiltmc/syntaxpain), a minified version of the library I made so we could more easily maintain enigma-swing.

**Review by ix0rai — CHANGES_REQUESTED** [review]

File: enigma-swing/build.gradle [hunk 3]
```diff
@@ -1,23 +1,36 @@
...
 dependencies {
 	implementation project(':enigma')
 	implementation project(':enigma-server')
 
 	implementation 'net.sf.jopt-simple:jopt-simple:6.0-alpha-3'
-	implementation 'com.formdev:flatlaf:2.6'
-	implementation 'com.formdev:flatlaf-extras:2.6' // for SVG icons
-	implementation 'de.sciss:syntaxpane:1.2.1'
+	implementation 'com.formdev:flatlaf:3.2.1'
+	implementation 'com.formdev:flatlaf-extras:3.2.1' // for SVG icons
```
Comment (ix0rai) [inline enigma-swing/build.gradle:19]:
```suggestion
	implementation 'com.formdev:flatlaf-extras:3.2.5' // for SVG icons
```

File: enigma-swing/build.gradle [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
+
 dependencies {
 	implementation project(':enigma')
 	implementation project(':enigma-server')
 
 	implementation 'net.sf.jopt-simple:jopt-simple:6.0-alpha-3'
-	implementation 'com.formdev:flatlaf:2.6'
-	implementation 'com.formdev:flatlaf-extras:2.6' // for SVG icons
-	implementation 'de.sciss:syntaxpane:1.2.1'
+	implementation 'com.formdev:flatlaf:3.2.1'
```
Comment (ix0rai) [inline enigma-swing/build.gradle:18]:
```suggestion
	implementation 'com.formdev:flatlaf:3.2.5'
```

File: enigma-swing/build.gradle [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
+	implementation 'com.formdev:flatlaf-extras:3.2.1' // for SVG icons
+	implementation 'de.sciss:syntaxpane:1.3.0'
 	implementation 'com.github.lukeu:swing-dpi:0.10'
 	implementation 'org.drjekyll:fontchooser:2.5.2'
+
+	flatLafNatives.forEach {
+		implementation 'com.formdev:flatlaf:3.2.1:' + it
```
Comment (ix0rai) [inline enigma-swing/build.gradle:25]:
```suggestion
		implementation 'com.formdev:flatlaf:3.2.5:' + it
```

**Review by ix0rai — CHANGES_REQUESTED** [review]
flatlaf 3.2.5

File: enigma-swing/build.gradle (hunk 2 above)
Comment (modmuss50) [reply enigma-swing/build.gradle:20]: Yeah, ill just stick with the current version. Is there a bug report open for this issue?

Comment (NebelNidas) [reply enigma-swing/build.gradle]: Yup, it's here: <https://codeberg.org/sciss/SyntaxPane/issues/38>

**Pull request merged**